web/src/engine/websites/MangaHot.ts

```diff
@@ -0,0 +1,193 @@
+﻿import { Tags } from '../Tags';
+import icon from './MangaHot.webp';
+import { Chapter, DecoratableMangaScraper, Manga, type MangaPlugin, Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS, FetchJSON, } from '../platform/FetchProvider';
+import type { Priority } from '../taskpool/DeferredTask';
+import DeScramble from '../transformers/ImageDescrambler';
+
+type APIResult<T> = {
+    result: null | T
+}
+
+type PageResult = {
+    [id: string]: string
+}
+
+function MangaExtractor(element: HTMLAnchorElement) {
+    return {
+        id: new URL(element.href).searchParams.get('work_code'),
+        title: element.querySelector('div.work_name').textContent.trim()
+    };
+
+}
+
+@Common.MangasSinglePageCSS('/ranking', 'div.ranking a.parent', MangaExtractor)
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('mangahot', 'マンガほっと (Manga Hot)', 'https://web.mangahot.jp', Tags.Media.Manga, Tags.Language.Japanese, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/works/detail.php\\?work_code=[^/]+`).test(url);
```

Comment:
Regex will not work for valid url such as `https://web.mangahot.jp/works/detail.php?page=7&work_code=solo-leveling`.
It is not always appropriate to use a regular expression, there are other methods as well e.g.,
```typescript
const uri = new URL(url);
return uri.origin === this.URI.origin && uri.pathname === '/works/detail.php' && uri.searchParams.has('work_code');
```